src/tests/tx_settlement.rs

```diff
@@ -138,3 +138,71 @@ async fn test_full_settlement_flow() -> Result<()> {
 
     Ok(())
 }
+
+#[test_log::test(tokio::test)]
+async fn test_contract_upgrade() -> Result<()> {
+    let mut builder = NodeIntegrationCtxBuilder::new().await;
+    let rest_url = builder.conf.rest.clone();
+    let mut node_modules = builder.build().await?;
```

Comment:
is it possible to call it "node" please :D :D :D my brain can't deal with npm's "node_modules" nightmare XD

and it makes sense to write `node.wait_for_n_blocks(1).await?;`